code/modules/mob/living/living.dm

```diff
@@ -952,4 +952,17 @@
 
 //used in datum/reagents/reaction() proc
 /mob/living/proc/get_permeability_protection()
-	return 0
\ No newline at end of file
+	return 0
+
+/mob/living/verb/sniff()
+
+	set name = "Sniff"
+	set category = "IC"
+
+	if(istype(wear_mask, /obj/item/clothing/mask))//can't smell things with a mask on your face
```

Comment:
Probably want a check for if the mask covers your nose or not, so bandanas or lowered masks don't interfere with your ability to smell